src/Rest.php

```diff
@@ -84,4 +87,17 @@ public function userinfo() {
 		$this->server->handleUserInfoRequest(OAuth2\Request::createFromGlobals())->send();
 		exit;
 	}
+
+	public function is_consent_needed() : bool {
```

Comment:
I think the fact that this code, which is unrelated to Rest, is a bit of a smell, though I do understand why. I think one way to move this code elsewhere could be to define an "authorization was granted" hook (I guess `action`?), and then listen to that hook elsewhere (outside `Rest.php`).

There's also some things I don't understand about the current architecture of this plugin, which are unrelated to this PR, but I think could in fact be the root of the "problem".

Namely, I'm having some trouble understanding why the code that handles `openid-connect/authenticate` is not under `Rest.php` but is instead in `OpenIDConnectServer.php`.

I think it would make sense to re-evaluate the architecture of the plugin at some point.

Reply:
Yeah! I didn't feel great about putting that there but it was this or defining a helper class with static functions. So, for now, I did it this way to atleast get it functional.

> Namely, I'm having some trouble understanding why the code that handles openid-connect/authenticate is not under Rest.php but is instead in OpenIDConnectServer.php.

Yeah, its a little weird to have it that way, but I suppose its done so that REST endpoints are not serving a html page and handling submission of that page.

> I think it would make sense to re-evaluate the architecture of the plugin at some point.

Yep, agree! I am going to merge it for now and we can revisit later to work out the wrinkles.